### PR TITLE
GEODE-7837: Java client support SNI proxying; config via system property

### PR DESCRIPTION
**Pull request by Bill**
[GEODE-7837](https://issues.apache.org/jira/browse/GEODE-7837)

Second try at this PR—now with more automated testing! `ClientSNIAcceptanceTest.java` runs a Geode cluster in a Docker container w/ an SNI proxy (haproxy) in a container out front.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-assembly/src/acceptanceTest/java/org/apache/geode/client/ClientSNIAcceptanceTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    gemFireProps.setProperty(SSL_TRUSTSTORE_PASSWORD, "geode");
+    // gemFireProps.setProperty(SSL_ENDPOINT_IDENTIFICATION_ENABLED, "true");
+
+    ClientCache cache = new ClientCacheFactory(gemFireProps)
+        .addPoolLocator(
+            "localhost",
+            55221)
+        .create();
+    Region<String, String> region =
+        cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
```
Comment (bschuchardt) [inline geode-assembly/src/acceptanceTest/java/org/apache/geode/client/ClientSNIAcceptanceTest.java]: You should probably use an empty region here.  With a CACHING_PROXY your get() operation is going to find the value locally.

Comment (Bill) [reply geode-assembly/src/acceptanceTest/java/org/apache/geode/client/ClientSNIAcceptanceTest.java]: thanks! fixed

**Comment by Bill** [comment]
that LGTM failure appears to be a false negative. I manually initiated a re-run just now.

**Comment by Bill** [comment]
This branch has been merged into @upthewaterspout 's https://github.com/apache/geode/pull/4830

**Pull request closed without merging**